src/components/content/credential/AddCredential.tsx

```diff
@@ -0,0 +1,300 @@
+        setCredentialVariableList([]);
+    };
+
+    const handleCredentialTypeSelect = (type: CredentialVariables.type) => {
+        setCurrentType(type);
+        form.setFieldsValue({ variables: [] });
+    };
+
+    useEffect(() => {
+        if (currentCsp === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialTypesByCsp(currentCsp)
+            .then((resp) => {
+                if (resp.length > 0) {
+                    setCredentialTypeList(resp as CredentialVariables.type[]);
+                } else {
+                    setCredentialTypeList([]);
+                }
+            })
+            .catch((error: Error) => {
+                setCredentialTypeList([]);
+                console.log(error);
+            });
+
+        if (currentType === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialCapabilitiesByCsp(currentCsp, currentType)
+            .then((resp) => {
+                if (resp.length > 0) {
+                    const credentialVariables: CredentialVariables = resp[0] as CredentialVariables;
+                    const credentialVariableList: CredentialVariable[] = [];
+                    credentialVariables.variables.forEach((credentialVariable) => {
+                        credentialVariable.value = '';
+                        credentialVariableList.push(credentialVariable);
+                    });
+                    setCredentialVariableList(credentialVariableList);
+                    form.setFieldsValue({ variables: credentialVariableList });
+                } else {
+                    setCredentialVariableList([]);
+                    form.setFieldsValue({ variables: [] });
+                }
+            })
+            .catch((error: Error) => {
+                setCredentialVariableList([]);
+                form.setFieldsValue({ variables: [] });
+                console.log(error);
+            });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [currentCsp, currentType]);
+
+    function setVariablesValue(index: number, e: ChangeEvent<HTMLInputElement>) {
+        credentialVariableList[index].value = e.target.value;
+        form.setFieldsValue({ variables: credentialVariableList });
+    }
+
+    const columns: ColumnsType<CredentialVariable> = [
+        {
+            title: 'name',
+            dataIndex: 'name',
+            render: (value, record) =>
+                record.isMandatory ? (
+                    <div>
+                        <span className={'add-credential-from-variables-value'}>*</span>
+                        {value}
+                    </div>
+                ) : (
+                    <div>{value}</div>
+                ),
+        },
+        {
+            title: 'value',
+            dataIndex: 'value',
+            render: (value, record, index) =>
+                record.isMandatory ? (
+                    <Form.Item
```

Comment:
if the variable name is a little long, then the field name is moving to next line. Actually modal is small and the page hs still space. We must try to keep it always in one single line. 

![image](https://github.com/eclipse-xpanse/xpanse-ui/assets/54129659/bcac66ea-e222-4772-850a-4265ec306cee)